tests/virt_autotest/validate_system_health.pm

```diff
@@ -0,0 +1,79 @@
+# SUSE's openQA tests
+#
+# Copyright 2012-2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Do a basic examination to the host and guests after tests run
+# Maintainer: Julie CAO <JCao@suse.com>, qe-virt@suse.de
+
+use base "virt_feature_test_base";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use ipmi_backend_utils qw(reconnect_when_ssh_console_broken);
+use virt_autotest::common;
+use virt_autotest::utils qw(is_xen_host start_guests check_port_state check_host_health check_guest_health);
+use virt_utils qw(collect_host_and_guest_logs);
+
+sub prepare_run_test {
+    reconnect_when_ssh_console_broken unless defined(script_run("rm -f /root/{commands_history,commands_failure}", die_on_timeout => 0));
+    script_run("history -c");
+}
+
+sub run_test {
+
+    my $self = shift;
+    my %health_status = ();
+    $health_status{host} = check_host_health;
+    foreach my $guest (keys %virt_autotest::common::guests) {
+        script_run("virsh start $guest", die_on_timeout => 0);
+        if (script_retry("nmap $guest -PN -p ssh | grep open", delay => 2, retry => 60, die => 0) == 0) {
+            $health_status{$guest} = check_guest_health($guest);
+            if ($health_status{$guest} eq 'fail' || get_var('FORCE_UPLOAD_LOGS')) {
```

Comment:
If you are going to use such general name ```FORCE_UPLOAD_LOGS```, then I think this variable can also be used in other scenarios to upload logs anyway. Otherwise, use a more specific name.

Reply:
`FORCE_UPLOAD_LOGS` can be used in any test suite which calls validate_system_health. Only chained tests don't call validate_system_health, ie. `FORCE_UPLOAD_LOGS` cannot be used in chained tests(with `START_DIRECTLY_AFTER_TEST`).